include/directx/d3dx12.h

```diff
@@ -4037,6 +4037,683 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+                m_dProtectedResourceSessionTypeCount[NodeIndex].Count = 0;
+            }
+
+            // Special procedure to initialize local protected resource session types structs
+            // Must wait until session type count initialized
+            QueryProtectedResourceSessionTypes(NodeIndex, m_dProtectedResourceSessionTypeCount[NodeIndex].Count);
+        }
+
+        // Initialize features that requires highest version check
+        if (FAILED(m_hStatus = QueryHighestShaderModel())) {
+            return m_hStatus;
+        }
+
+        if (FAILED(m_hStatus = QueryHighestRootSignatureVersion())) {
+            return m_hStatus;
+        }
+
+        // Initialize Feature Levels data
+        if (FAILED(m_hStatus = QueryHighestFeatureLevel())) {
+            return m_hStatus;
+        }
+        
+        return m_hStatus;
+    }
+
+    HRESULT GetStatus() const
+    {
+        return m_hStatus;
+    }
+
+#ifdef FEATURE_SUPPORT_VARIANT_OPT_IN
+    static std::variant<CD3DX12FeatureSupport, HRESULT> Create(ID3D12Device* pDevice)
+    {
+        CD3DX12FeatureSupport features;
+        HRESULT result = features.Init(pDevice);
+        if (FAILED(result)) {
+            return result;
+        }
+
+        return features;
+    }
+#else
+    static CD3DX12FeatureSupport Create(ID3D12Device* pDevice)
+    {
+        CD3DX12FeatureSupport features;
+        features.Init(pDevice);
+        return features;
+    }
+#endif
+
+    // 0: D3D12_OPTIONS
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, DoublePrecisionFloatShaderOps);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, OutputMergerLogicOp);
+    FEATURE_SUPPORT_GET(D3D12_SHADER_MIN_PRECISION_SUPPORT, m_dOptions, MinPrecisionSupport);
+    FEATURE_SUPPORT_GET(D3D12_TILED_RESOURCES_TIER, m_dOptions, TiledResourcesTier);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_BINDING_TIER, m_dOptions, ResourceBindingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, PSSpecifiedStencilRefSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, TypedUAVLoadAdditionalFormats);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, ROVsSupported);
+    FEATURE_SUPPORT_GET(D3D12_CONSERVATIVE_RASTERIZATION_TIER, m_dOptions, ConservativeRasterizationTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, StandardSwizzle64KBSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, CrossAdapterRowMajorTextureSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_HEAP_TIER, m_dOptions, ResourceHeapTier);
+
+    // Special procedure for handling caps that is also part of other features
+    D3D12_CROSS_NODE_SHARING_TIER CrossNodeSharingTier() const
+    {
+        if (m_dCrossNode.SharingTier > D3D12_CROSS_NODE_SHARING_TIER_NOT_SUPPORTED)
+        {
+            return m_dCrossNode.SharingTier;
+        }
+        else
+        {
+            return m_dOptions.CrossNodeSharingTier;

```

Comment:
Some caps such as `CrossNodeSharingTier` appears in more than one feature data struct. While one of these structs are not recognized by the runtime, the other feature may still be available and provides support for the cap. Fortunately in most cases the two caps get their information from the same variable/function. I added a check for these features so they will check for both caps and return the one that's properly initialized.